src/silly_sprites/opengl/gl_window.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h>
+#include <spdlog/spdlog.h>
+#include <tl/expected.hpp>
+#include "gl_error.hpp"
+
+namespace sly::gl {
+	
+	class Window final {
+	private:
+		GLFWwindow* m_window;
+
+    public:
+		Window(GLFWwindow* window);
+
+		[[nodiscard]] static tl::expected<Window,WindowError> create(int width, int height);
+		static void destroy();
+
+		[[nodiscard]] GLFWwindow* get() const;
```

Comment:
I think this function should not be "open to the public". Maybe use introduce `friend` functions or classes if they really need to access this.
Reasoning: The fact that you use a `GLFWWindow*` internally is an implementation detail. This should not be visible to the outside.